### PR TITLE
CMakeLists.txt: integrate googletest as submodule

### PR DESCRIPTION
**Pull request by hnbrds**
This PR integrates googletest to the project, by adding googletest project as submodule

### CODE REVIEW

**Comment by 9rum** [comment]
@hnbrds Thanks for filing this PR. Have you verified it in your local?

**Comment by hnbrds** [comment]
@9rum For verification, I've wrote a sample test code ``test.cc``.
```cc
#include "gtest/gtest.h"

TEST(Foo, Sum) {
  EXPECT_EQ(2, 1+1);
}
```

Compiled as an executable file,
image below shows the execution result and it seems to work fine.

스크린샷 2024-01-22 오후 4 04 35

also works with ctest

스크린샷 2024-01-22 오후 4 32 38

**Comment by hnbrds** [comment]
1. Instead of adding all tests respectively with `add_test()`,
`gtest_discover_tests()` automatically discovers tests included in the executable.

```
root@622569c8ba94:/workspace/build# ctest .
Test project /workspace/build
    Start 1: Foo.Sum
1/1 Test #1: Foo.Sum ..........................   Passed    0.01 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.01 sec
```

2. Added option to make a decision whether to build test files; note that the option is set to `OFF`.

**Comment by hnbrds** [comment]
rather than using additional option `MAKO_BUILD_TESTING`, used automatically generated option `BUILD_TESTING` by CTest.

(https://cmake.org/cmake/help/latest/module/CTest.html)
As `BUILD_TESTING` option is enabled by default,
explicitly state option `-DBUILD_TESTING=OFF` in order to disable test file build.

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,4 +47,10 @@ find_package(
   REQUIRED
   COMPONENTS context)
 
+include CTest
```
Comment (9rum) [inline CMakeLists.txt]: Please change this line to ``include(CTest)``.

**Review by 9rum — CHANGES_REQUESTED** [review]

**Review by 9rum — APPROVED** [review]

**Pull request merged**